javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,356 @@
+        } else {
+          head.remValue()
+        }
+        _size -= 1
+        oldHead
+      }
+    }
+  }
+
+  def removeLast(): E = {
+    if (_size <= 0) throw new NoSuchElementException()
+    else {
+      if (last.isEmpty) {
+        val oldLast = head.getValue().get.inner
+        head.remValue()
+        head.remPrev()
+        head.remNext()
+        _size -= 1
+        oldLast
+      } else {
+        val newLast = last.getPrev()
+        val oldLast = last.getValue().get.inner
+        if (!newLast.isEmpty) {
+          last = newLast.get
+          last.remNext()
+        } else {
+          last.remValue()
+        }
+        _size -= 1
+        oldLast
+      }
+    }
+  }
+
+  def addFirst(e: E): Unit = {
+    if (head.isEmpty) {
+      head.setValue(Box(e))
+      head.remNext()
+      head.remPrev()
+    } else if (last.isEmpty) {
+      last.setValue(head.getValue().get)
+      last.setPrev(head)
+      last.remNext()
+      head.setValue(Box(e))
+      head.setNext(last)
+      head.remPrev()
+    } else {
+      val node = new Node(Some(Box(e)))
+      head.setPrev(node)
+      node.setNext(head)
+      head = node
+    }
+    _size += 1
+  }
+
+  def addLast(e: E): Unit = {
+    if (last.isEmpty) {
+      last.setValue(Box(e))
+    } else if (head.isEmpty) {
+      head.setValue(last.getValue().get)
+      head.setNext(last)
+      head.remPrev()
+      last.setValue(Box(e))
+      last.setPrev(head)
+      last.remNext()
+    }else {
+      val node = new Node(Some(Box(e)))
+      last.setNext(node)
+      node.setPrev(last)
+      last = node
+    }
+    _size += 1
+  }
+
+  override def contains(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+
+    @tailrec
+    def check(n: Node[Box[E]]): Boolean = {
+      if (n.isEmpty) false else {
+        if (n.getValue().get.equals(boxed)) true
+        else if (n.getNext().isEmpty) false
+        else check(n.getNext().get)
+      }
+    }
+
+    if (head.getValue().isEmpty)
+      check(last)
+    else
+      check(head)
+  }
+
+  override def size(): Int =
+    _size
+
+  override def add(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+    val iter = listIterator()
+    var changed = false
+    while (iter.hasNext()) {
+      if (Box(iter.next()).equals(boxed)) {
+        iter.remove()
+        changed = true
+      }
+    }
+
+    changed
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    var changed = iter.hasNext()
+    while (iter.hasNext())
+      addLast(iter.next())
+
+    changed
+  }
+
+  override def clear(): Unit = {
+    head = new Node(None)
+    last = new Node(None)
+    _size = 0
+  }
+
+  override def get(index: Int): E = {
+    checkIndexInBounds(index)
+
+    throw new Exception("TBD")
+  }
+
+  override def set(index: Int, element: E): E = {
+    add(index, element)
+    remove(index+1)
```

Comment:
Actually the `set` from `AbstractList` is also doing two traversal. You should improve it using a `ListIterator` and the drop the override.